backend/src/controllers/middlewaresControllers/createUserController/updatePassword.js

```diff
@@ -3,52 +3,51 @@ const bcrypt = require('bcryptjs');
 const { generate: uniqueId } = require('shortid');
 
 const updatePassword = async (userModel, req, res) => {
-  const UserPassword = mongoose.model(userModel + 'Password');
-
-  const reqUserName = userModel.toLowerCase();
-  const userProfile = req[reqUserName];
-
-  let { password } = req.body;
-
-  if (password.length < 8)
-    return res.status(400).json({
-      msg: 'The password needs to be at least 8 characters long.',
-    });
-
-  // Find document by id and updates with the required fields
-
-  const salt = uniqueId();
-
-  const passwordHash = bcrypt.hashSync(salt + password);
-
-  const UserPasswordData = {
-    password: passwordHash,
-    salt: salt,
-  };
-
-  const resultPassword = await UserPassword.findOneAndUpdate(
-    { user: req.params.id, removed: false },
-    { $set: UserPasswordData },
-    {
-      new: true, // return the new result instead of the old one
+  try {
+    const UserPasswordModel = mongoose.model(userModel + 'Password');
+    const reqUserName = userModel.toLowerCase();
+    const user = req[reqUserName];
+    const { password } = req.body;
+
+    if (password.length < 8) {
+      return res.status(400).json({
+        success: false,
```

Comment:
The success attribute in the response is unnecessary as we can determine the status of the response from the response code